packages/server/src/mL/index.ts

```diff
@@ -35,14 +36,24 @@ export default class MLService {
 
     await this.docker.reset();
 
-    this.dockerState = DockerState.TrainPull;
-    await this.docker.pullImages(Object.values(Trainer.images));
-
-    this.dockerState = DockerState.ExportPull;
-    await this.docker.pullImages(Object.values(Exporter.images));
-
-    this.dockerState = DockerState.TestPull;
-    await this.docker.pullImages(Object.values(Tester.images));
+    https
+      .get("https://hub.docker.com/", async (res) => {
+        if (res.statusCode === 200) {
+          this.dockerState = DockerState.TrainPull;
+          await this.docker.pullImages(Object.values(Trainer.images));
+
+          this.dockerState = DockerState.ExportPull;
+          await this.docker.pullImages(Object.values(Exporter.images));
+
+          this.dockerState = DockerState.TestPull;
+          await this.docker.pullImages(Object.values(Tester.images));
+        } else {
```

Comment:
Should then check to see if images exist within this else. If they don't, Axon should request internet be found. Should not be able to do anything until internet is found. Should have a "retry" button to have axon look for internet again.